core-bundle/src/Security/Authentication/Token/TokenChecker.php

```diff
@@ -56,6 +57,24 @@ public function hasBackendUser(): bool
         return null !== $token && $token->getUser() instanceof BackendUser;
     }
 
+    /**
+     * Checks if a user has a two-factor token (in 2FA progess).
+     */
+    public function hasTwoFactorToken(): bool
+    {
+        $token = null;
+
+        if ($this->hasFrontendUser()) {
+            $token = $this->getToken(FrontendUser::SECURITY_SESSION_KEY);
+        }
+
+        if ($this->hasBackendUser()) {
```

Comment:
This won't work, as the user can have both a front end and back end user logged in.

Reply:
How do you want to fix it? This case only occurs, if you stuck in the 2FA process (logged in, but not confirmed second factor yet) at the same time in frontend and backend.